compiler/test/test_end_to_end.re

```diff
@@ -1053,6 +1054,17 @@ let match_tests = [
      }",
     "79",
   ),
+  // Constant patterns
+  t(
+    "constant_match_1",
+    "match (1/3) { | 5 => false | 2/6 => true | _ => false }",
+    "true",
+  ),
+  t(
+    "constant_match_2",
+    "match (('foo', 5, false)) { | ('bar', 5, false) => false | ('foo', _, true) => false | ('foo', _, false) => true | _ => false }",
+    "true",
+  ),
```

Comment:
I'd like to add some tests that match constants with additional guards added.

Reply:
Good call.